File: grim/src/main/java/io/turbo/clientBlock/config/BlockedClients.java
package io.turbo.clientBlock.config;

import de.exlll.configlib.Comment;
import de.exlll.configlib.Configuration;
import de.exlll.configlib.YamlConfigurations;
import io.turbo.clientBlock.ClientBlock;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.io.File;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

@Getter
@NoArgsConstructor
@Configuration
public class BlockedClients {
    private static BlockedClients instance;

    @Comment("List of Blocked Clients")
    List<String> blockedClients = new ArrayList<>();

    public void save() {
        YamlConfigurations.save(new File(ClientBlock.getInstance().getDataFolder(), "blockedclients.yml").toPath(), BlockedClients.class, this);
    }

    public static void reload() {
        Path path = new File(ClientBlock.getInstance().getDataFolder(), "blockedclients.yml").toPath();
        instance = YamlConfigurations.update(path, BlockedClients.class);
    }


    public static BlockedClients i() {
        if (instance != null) {
            return instance;
        }

        return instance = YamlConfigurations.update(new File(ClientBlock.getInstance().getDataFolder(), "bounties.yml").toPath(), BlockedClients.class);
    }

    public static BlockedClients load() {
        Path path = new File(ClientBlock.getInstance().getDataFolder(), "blockedclients.yml").toPath();
        return YamlConfigurations.load(path, BlockedClients.class);
    }
}

File: common-kotlin/src/main/kotlin/io/turbo/clientblockkotlin/config/BlockedClients.kt
package io.turbo.clientblockkotlin.config

import de.exlll.configlib.Comment
import de.exlll.configlib.Configuration
import de.exlll.configlib.YamlConfigurations
import io.turbo.clientblockkotlin.ClientBlock
import lombok.Getter
import lombok.NoArgsConstructor
import java.io.File
import java.nio.file.Path

@Getter
@NoArgsConstructor
@Configuration
class BlockedClients {
    @Comment("List of Blocked Clients")
    var blockedClients: MutableList<String?> = ArrayList<String?>()

    fun save() {
        YamlConfigurations.save<BlockedClients?>(
            File(
                ClientBlock.i()?.dataFolder, "blockedclients.yml"
            ).toPath(), BlockedClients::class.java, this
        )
    }

    companion object {
        private var instance: BlockedClients? = null

        fun reload() {
            val path: Path = File(
                ClientBlock.i()?.dataFolder,
                "blockedclients.yml"
            ).toPath()
            instance = YamlConfigurations.update<BlockedClients?>(path, BlockedClients::class.java)
        }


        fun i(): BlockedClients? {
            if (instance != null) {
                return instance
            }

            return YamlConfigurations.update<BlockedClients?>(
                File(
                    ClientBlock.i()?.dataFolder, "bounties.yml"
                ).toPath(), BlockedClients::class.java
            ).also { instance = it }
        }

        fun load(): BlockedClients? {
            val path: Path = File(
                ClientBlock.i()?.dataFolder,
                "blockedclients.yml"
            ).toPath()
            return YamlConfigurations.load<BlockedClients?>(path, BlockedClients::class.java)
        }
    }
}
